pkg/providers/v2/instances.go

```diff
@@ -226,6 +226,20 @@ func nodeAddressesForInstance(instance *ec2.Instance) ([]v1.NodeAddress, error)
 				})
 			}
 		}
+
+		for _, privateIP := range networkInterface.Ipv6Addresses {
```

Comment:
The behavior in the v1 provider is a bit different, we only utilize the first entry in this array: https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/providers/v1/aws.go#L1574

I don't have all the context as to why that was done.

Reply:
I'm happy to swap to that behaviour instead if that'll get the PR landed.


> I don't have the context as to why that was done.

My guess is that because the first IP cannot be modified (I vaguely recall this from somewhere?); but the others can at runtime by modifying the ENI; and the author of v1 was trying to keep it as a static list?

Reply:
This is kind of related to: https://github.com/kubernetes/cloud-provider-aws/issues/656

The VPC CNI will allocate and utilize additional ENI's for pod IP's. The pod IP's are filtered out from the Node's addresses when the `provided-node-ip` annotation is present: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/node/helpers/address.go#L115-L120

But ideally we wouldn't add Pod IP's in the first place. I **think** we should only be adding the first address on the first ENI

@tzneal WDYT?

Reply:
I was wondering if we should just use the PrivateDnsName/PrivateIpAddress from the describe-instances output and ignore ENIs.  Anything we can do will break someone, maybe a flag to control what we do?

Reply:
Hm yeah I think you're right; that's what we end up with in practice on EKS anyway. But I'm not sure how this would impact other CNI's, cluster addons, etc; so making it configurable is a safe bet.